examples/matrix_multiplier/tests/test_matrix_multiplier.py

```diff
@@ -0,0 +1,181 @@
+import os
+from random import getrandbits
+
+import cocotb
+from cocotb.binary import BinaryValue
+from cocotb.clock import Clock
+from cocotb.monitors import BusMonitor
+from cocotb.regression import TestFactory
+from cocotb.scoreboard import Scoreboard
+from cocotb.triggers import RisingEdge, ReadOnly
+
+NUM_SAMPLES = int(os.environ.get('NUM_SAMPLES', 3000))
+DATA_WIDTH = int(os.environ['DATA_WIDTH'])
+A_ROWS = int(os.environ['A_ROWS'])
+B_COLUMNS = int(os.environ['B_COLUMNS'])
+A_COLUMNS_B_ROWS = int(os.environ['A_COLUMNS_B_ROWS'])
+
+
+class MatMultMonitor(BusMonitor):
+
+    _signals = ["A", "B", "i_valid", "C", "o_valid"]
```

Comment:
It seems like you could just instance a data/valid monitor at the input and an output, set the software model as the callback on the input monitor, and have that model populate the expected data list. But I also don't know anything about the included base classes. They seem hard to use.